catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrCatalogProvider.java

```diff
@@ -184,7 +187,15 @@ public boolean isAvailable() {
 
   @Override
   public boolean isAvailable(SourceMonitor callback) {
-    return isAvailable();
+    solr.isAvailable( // first register the callback
```

Comment:
✏️ Unless I'm missing something and this will only be called once, I recommend...
1. Change the underlying implementation of your `listeners` collection to be a set (`LinkedHashSet` if insertion order matters for some reason, but I can't see why).
2. Don't just use lambdas but make explicit implementations of your `Listener` interface so you can add `equals`/`hashcode` methods.

Your list is going to grow unbounded, with duplicate listeners in it.

Reply:
Based on our outside discussions, I am leaving it using lambdas which after all are simply interfaces which could still be implemented on their own. I will change the underlying implementation to use a set and I will change this code to actually extends the class and base its equal and hashcode on the specified monitor callback object. 
The lambdas do add in place where the listener is only added one a simple touch. But I agree here that it doesn't.
⭕️

Reply:
👍